sphinx/domains/c.py

```diff
@@ -3254,27 +3234,8 @@ def handle_signature(self, sig: str, signode: TextElement) -> ASTDeclaration:
 
         parser = DefinitionParser(sig, location=signode, config=self.env.config)
         try:
-            try:
-                ast = self.parse_definition(parser)
-                parser.assert_end()
-            except DefinitionError as eOrig:
-                if not self.env.config['c_allow_pre_v3']:
-                    raise
-                if self.objtype != 'type':
-                    raise
-                try:
-                    ast = self.parse_pre_v3_type_definition(parser)
-                    parser.assert_end()
-                except DefinitionError:
-                    raise eOrig
-                self.object_type = ast.objectType  # type: ignore
-                if self.env.config['c_warn_on_allowed_pre_v3']:
-                    msg = "{}: Pre-v3 C type directive '.. c:type:: {}' converted to " \
-                          "'.. c:{}:: {}'." \
-                          "\nThe original parsing error was:\n{}"
-                    msg = msg.format(RemovedInSphinx70Warning.__name__,
```

Comment:
Note this was bumped to 70 per request in https://github.com/sphinx-doc/sphinx/pull/10471#discussion_r888641864, though no version was ever published with the delay in deprecation so we are still OK to remove in 6.0.

A